### PR TITLE
Feature/egdl 2674

### PR DESCRIPTION
**Pull request by andreeapad**

### CODE REVIEW

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java [hunk 1]
```diff
@@ -36,4 +37,9 @@
       + "and p.modifiedTimestamp <= :instant order by p.modifiedTimestamp")
   Page<HousekeepingPath> findRecordsForCleanupByModifiedTimestamp(@Param("instant") LocalDateTime instant,
       Pageable pageable);
+
+  @Modifying
+  @Query(value = "delete from HousekeepingPath p where p.cleanupTimestamp < :instant "
```
Comment (abhimanyugupta07) [inline beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java:42]: Let's order it by `order by p.cleandupTimestamp` so we are deleting rows in order. I found it useful to see the progress while debugging.

File: beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/service/MetadataRepositoryCleanupService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import io.micrometer.core.annotation.Timed;
+
+import com.expediagroup.beekeeper.cleanup.service.RepositoryCleanupService;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+public class MetadataRepositoryCleanupService implements RepositoryCleanupService {
+
+  private final HousekeepingMetadataRepository housekeepingMetadataRepository;
+  private final int numberOfRetentionDays;
```
Comment (abhimanyugupta07) [inline beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/service/MetadataRepositoryCleanupService.java:34]: `retentionPeriodInDays`. sounds better ?

File: beekeeper-metadata-cleanup/src/test/java/com/expediagroup/beekeeper/metadata/cleanup/service/MetadataRepositoryCleanupServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  public void typical() {
+    MetadataRepositoryCleanupService repositoryCleanupService = new MetadataRepositoryCleanupService(housekeepingMetadataRepository,
+        2);
+    LocalDateTime now = LocalDateTime.of(2021, 11, 12, 10, 10);
+    repositoryCleanupService.cleanUp(now.toInstant(ZoneOffset.UTC));
+
+    LocalDateTime referenceTime = LocalDateTime.of(2021, 11, 10, 10, 10);
+    verify(housekeepingMetadataRepository).cleanUpOldDeletedRecords(referenceTime);
+  }
+}
```
Comment (abhimanyugupta07) [inline beekeeper-metadata-cleanup/src/test/java/com/expediagroup/beekeeper/metadata/cleanup/service/MetadataRepositoryCleanupServiceTest.java:45]:
```suggestion
}

```

**Review by abhimanyugupta07 — COMMENTED** [review]
Looks good. Left some small suggestions.

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java (hunk 1 above)
Comment (andreeapad) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java:42]: ok

Comment (andreeapad) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java:42]: tried to add the order by, but the tests fail cause apparently the queries cannot be validated 🤔

Comment (abhimanyugupta07) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java:42]: hmm.. how is it working with the getPathsToBeCleaned() query ?

Comment (abhimanyugupta07) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java:42]: Discussed offline. This is not a blocker. So, I will go ahead and approve this PR.

File: beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  private static final Logger log = LoggerFactory.getLogger(RepositoryCleanupScheduler.class);
+
+  private final RepositoryCleanupService cleanupService;
+
+  @Autowired
+  public RepositoryCleanupScheduler(RepositoryCleanupService cleanupService) {
+    this.cleanupService = cleanupService;
+  }
+
+  @Scheduled(cron = "${properties.repository-scheduler-cron}")
```
Comment (abhimanyugupta07) [inline beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java:38]: `deleted-paths-cleanup-cron` is that better ?

Comment (andreeapad) [reply beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java:38]: hmm, yeah, but it's going to be used for both paths and metadata 🤔

**Review by patduin — APPROVED** [review]
nice

File: beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java (hunk 4 above)
Comment (abhimanyugupta07) [reply beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java:38]: just `old-data-cleanup-cron` ?

Comment (abhimanyugupta07) [reply beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java:38]: README will need an update

Comment (andreeapad) [reply beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/service/RepositoryCleanupScheduler.java:38]: yess, ok

File: README.md [hunk 5]
```diff
@@ -255,6 +255,8 @@ To access an endpoint when running in a Docker container, the port must be publi
 | `cleanup-page-size`                 | No       | Number of rows that should be processed in one page. Default value is `500`. |
 | `dry-run-enabled`                   | No       | Enable to simply display the deletions that would be performed, without actually doing so. Default value is `false`. |
 | `scheduler-delay-ms`                | No       | Amount of time (in milliseconds) between consecutive cleanups. Default value is `300000` (5 minutes after the previous cleanup completes). |
+| `old-data-cleanup-cron`             | No       | Cron expression which sets the schedule for the cleanup of old rows in the `housekeeping_path` table. Default is `0 0 13 * * ?` (every day at 1pm). |
+| `old-data-retention-period-days`    | No       | Number of days to keep old rows in the `housekeeping_path` table after their corresponding data is deleted. Default is `182` (6 months). |
```
Comment (abhimanyugupta07) [inline README.md:259]: Also for metadata table, right ?

Comment (andreeapad) [reply README.md:259]: it's below

Comment (abhimanyugupta07) [reply README.md:259]: 🤦

**Review by abhimanyugupta07 — APPROVED** [review]

**Pull request merged**